mkosi

```diff
@@ -1054,8 +1054,27 @@ def invoke_dnf(args: CommandLineArguments, workspace: str, repositories: List[st
     with mount_api_vfs(args, workspace):
         run(cmdline, check=True)
 
+    if cleanup:
+        # If dnf is not installed, there doesn't seem to be much use in keeping
+        # the dnf metadata, since it's not usable from within the image anyway.
+        # Similarly for rpm, remove rpm metadata if rpm is not installed.
+        keep_dnf_data = 'dnf' in packages
+        keep_rpm_data = keep_dnf_data or 'rpm' in packages
+
+        if not keep_dnf_data:
+            print_step('Cleaning dnf metadata...')
+            remove_glob(root + '/var/lib/dnf',
+                        root + '/var/log/dnf.*',
+                        root + '/var/log/hawkey.*',
+                        root + '/var/cache/dnf',
+                        root + '/etc/yum/repos.d',
+                        root + '/etc/dnf')
```

Comment:
what's in these /etc directories? why do they exist at all if dnf is not installed?

I mean, I can totally see that dnf creates the dirs in /var while running even if it is not installed in the image itself, but what about /etc?

Reply:
Hmm, I guess it's better to keep the files in /etc. They won't be used be anything, but they are very small, so it doesn't much either way. I'll trim the list to `/var` only.